controllers/webhook/upstream_services.go

```diff
@@ -1,114 +1,107 @@
 package webhook
 
 import (
-	"bytes"
 	"context"
-	"encoding/json"
-	"errors"
 	"fmt"
-	config "github.com/sky-uk/kfp-operator/apis/config/v1alpha5"
 	"github.com/sky-uk/kfp-operator/argo/common"
-	"io"
-	"net"
-	"net/http"
-	"strconv"
-	"time"
+	"google.golang.org/grpc/credentials/insecure"
+
+	config "github.com/sky-uk/kfp-operator/apis/config/v1alpha5"
+	pb "github.com/sky-uk/kfp-operator/triggers/run-completion-event-trigger/proto"
+	"google.golang.org/grpc"
+	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
 type UpstreamService interface {
-	call(ctx context.Context, ed EventData) error
+	call(ctx context.Context, ed common.RunCompletionEvent) error
 }
 
-type ArgoEventWebhook struct {
-	Upstream config.Endpoint
-	Client   *http.Client
+type GrpcNatsTrigger struct {
```

Comment:
In the controller side we dont know anything about what the underlying trigger is going to do so no need to say anything about NATS here now, it is just a gRPC trigger

Reply:
changed to be generic `GrpcTrigger`